Public/Src/Engine/Dll/Recovery/CorruptedMemosDbRecovery.cs

```diff
@@ -112,12 +112,7 @@ public override Possible<Unit> Recover()
             //         RenameMemoDb();
             //     }
             var corruptedMemosDbFile = PrepareCorruptedMemosDbBackupFile(cacheDirectory);
-            var result = FileUtilities.MoveFileAsync(memosDbFile, corruptedMemosDbFile, replaceExisting: true).Result;
-
-            if (!result)
-            {
-                return new Failure<string>(I($"Failed to rename '{memosDbFile}' to '{corruptedMemosDbFile}'"));

```

Comment:
Here you did change the behavior of `Possible<Unit> Recover()` to now throw instead of returning a Failure